src/types/infer.ts

```diff
@@ -0,0 +1,1596 @@
+                    def instanceof EnumDefinition ||
+                    def instanceof ContractDefinition)
+        ) as Array<StructDefinition | EnumDefinition | ContractDefinition>;
+
+        const eventDefs = defs.filter(
+            (def) => !externalOnly && def instanceof EventDefinition
+        ) as EventDefinition[];
+
+        const errorDefs = defs.filter(
+            (def) => !externalOnly && def instanceof ErrorDefinition
+        ) as ErrorDefinition[];
+
+        if (funs.length > 0) {
+            assert(
+                funs.length === defs.length,
+                "Unexpected both functions and others matching {0} in {1}",
+                name,
+                ctx
+            );
+
+            if (funs.length === 1) {
+                const res = this.funDefToType(funs[0]);
+
+                /**
+                 * @todo (Pavel) Consider checking if resolved function is externally accessible (external, public)
+                 */
+
+                return res;
+            }
+
+            return new FunctionLikeSetType(funs);
+        }
+
+        if (getters.length > 0) {
+            assert(
+                getters.length === defs.length,
+                "Unexpected both getters and others matching {0} in {1}",
+                name,
+                ctx
+            );
+
+            assert(getters.length === 1, "Unexpected overloading between getters for {0}", name);
+
+            return externalOnly
+                ? getters[0].getterFunType()
+                : this.variableDeclarationToTypeNode(getters[0]);
+        }
+
+        if (errorDefs.length > 0) {
+            assert(
+                errorDefs.length === defs.length,
+                "Unexpected both getters and others matching {0} in {1}",
+                name,
+                ctx
+            );
+
+            assert(
+                errorDefs.length === 1,
+                "Unexpected overloading between errorDefs for {0}",
+                name
+            );
+
+            return this.errDefToType(errorDefs[0]);
+        }
+
+        if (eventDefs.length > 0) {
+            assert(
+                eventDefs.length === defs.length,
+                "Unexpected both events and others matching {0} in {1}",
+                name,
+                ctx
+            );
+
+            if (eventDefs.length === 1) {
+                return this.eventDefToType(eventDefs[0]);
+            }
+
+            return new FunctionLikeSetType(eventDefs);
+        }
+
+        assert(typeDefs.length == 1, "Unexpected number of type defs {0}", name);
+
+        const def = typeDefs[0];
+        const fqName = getFQDefName(def);
+
+        return new TypeNameType(new UserDefinedType(fqName, def));
+    }
+
+    /**
+     * Infer the data location for the given `VariableDeclaration`. For local vars with
+     * solidity <=0.4.26 we infer the location from the RHS.
+     */
+    private inferVariableDeclLocation(decl: VariableDeclaration): DataLocation {
```

Comment:
This function is not yet removed in `src/types/utils.ts` and is slightly different.

Migrating `inferVariableDeclLocation` and `variableDeclarationToTypeNode` will hit consumers codebase. We should proceed here with care if we do not want to cause too much impact BC much.

Reply:
UPD: I see that we start to migrate functionality, so I guess we can dispute comment above.

Reply:
Good call to mark those deprecated

Reply:
@cd1m0 see https://github.com/ConsenSys/scribble/pull/190, especially https://github.com/ConsenSys/scribble/pull/190/commits/76b964cfe849ef1da56975e6680bab98ab6fe7e8. Both versions of `variableDeclarationToTypeNode` are currently used in some nodes of **solc-typed-ast** and in Scribble. The `inferVariableDeclLocation` is `private` in current PR, so I would also need investigate if it would have significant impact on Scribble.

I guess we should slowly migrate to `variableDeclarationToTypeNode` of `InferType` instead of pure function. But it would  impact both packages if we will do this right now.

Reply:
Thanks for pointing out. I will tweak the PR to make `InferType.variableDeclarationToTypeNode` public before pushing, and work on https://github.com/ConsenSys/scribble/pull/190 to make the codebase use the correct variableDeclarationToTypenode.